containers/jvm/clipper-spark/src/main/scala/ai/clipper/spark/Clipper.scala

```diff
@@ -0,0 +1,346 @@
+        p.save(modelPath)
+        // Because I'm not sure how to do it in the type system, check that
+        // the container is of the right type
+        try {
+          containerClass.newInstance.asInstanceOf[PipelineModelContainer]
+        } catch {
+          case e: ClassCastException => {
+            throw new IllegalArgumentException(
+              "Error: Container must be a subclass of PipelineModelContainer")
+          }
+        }
+        PipelineModelType
+      }
+      case _ =>
+        throw new IllegalArgumentException(
+          s"Illegal model type: ${model.getClass.getName}")
+    }
+    val jarPath = Paths.get(
+      containerClass.getProtectionDomain.getCodeSource.getLocation.getPath)
+    val copiedJarName = CONTAINER_JAR_FILE
+    println(s"JAR path: $jarPath")
+    System.out.flush()
+    Files.copy(jarPath, Paths.get(basePath, copiedJarName), REPLACE_EXISTING)
+    val conf =
+      ClipperContainerConf(containerClass.getName, copiedJarName, modelType)
+    getReplOutputDir(sc) match {
+      case Some(classSourceDir) => {
+        println(
+          "deployModel called from Spark REPL. Saving classes defined in REPL.")
+        conf.fromRepl = true
+        conf.replClassDir = Some(REPL_CLASS_DIR)
+        val classDestDir = Paths.get(basePath, REPL_CLASS_DIR)
+        FileUtils.copyDirectory(Paths.get(classSourceDir).toFile,
+                                classDestDir.toFile)
+      }
+      case None =>
+        println(
+          "deployModel called from script. No need to save additionally generated classes.")
+    }
+    Files.write(Paths.get(basePath, CLIPPER_CONF_FILENAME),
+                write(conf).getBytes,
+                CREATE)
+  }
+
+  private def getReplOutputDir(sc: SparkContext): Option[String] = {
+    sc.getConf.getOption("spark.repl.class.outputDir")
+  }
+
+  private[clipper] def loadSparkModel(
+      sc: SparkContext,
+      basePath: String): SparkModelContainer = {
+    val confString = Files
+      .readAllLines(Paths.get(basePath, CLIPPER_CONF_FILENAME),
+                    Charset.defaultCharset())
+      .get(0)
+
+    val conf = read[ClipperContainerConf](confString)
+    val classLoader = getClassLoader(basePath, conf)
+    val modelPath = Paths.get(basePath, MODEL_DIRECTORY).toString
+    println(s"Model path: $modelPath")
+
+    conf.modelType match {
+      case MLlibModelType => {
+        val model = MLlibLoader.load(sc, modelPath)
+        try {
+          val container = classLoader
+            .loadClass(conf.className)
+            .newInstance()
+            .asInstanceOf[MLlibContainer]
+          container.init(sc, model)
+          container.asInstanceOf[SparkModelContainer]
+        } catch {
+          case e: Throwable => {
+            e.printStackTrace
+            throw e
+          }
+
+        }
+      }
+      case PipelineModelType => {
+        val model = PipelineModel.load(modelPath)
+        val container = classLoader
```

Comment:
Consider defining a `createSparkContainer` method to avoid repeating code for each case